usb.c

```diff
@@ -1120,8 +1121,12 @@ int usb_context_scan(struct iio_scan_backend_context *ctx,
 		unsigned int interface = 0;
 
 		ret = libusb_open(dev, &hdl);
-		if (ret)
+		if (ret) {
+			if (ret != LIBUSB_ERROR_ACCESS)
+				perms = true;
 			continue;
+		}
+		perms = true;
```

Comment:
For me it makes no sense that some random other accessible device affects the result of this function. There is no correlation between other devices being accessible and the correct udev rules for IIO devices being installed.

The other issue here is that returning an error here will cause iio_scan_context_get_info_list() to fail, even though other contexts returned valid results. So this is pretty much a no-go.